packages/blade/src/components/Input/BaseInput/BaseInputWrapper.tsx

```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components';
+import type { ReactElement, ReactNode } from 'react';
+import { getInputBackgroundAndBorderStyles } from './baseInputStyles';
+import type { BaseInputProps } from './BaseInput';
+import { getPlatformType } from '~utils';
+import Box from '~components/Box';
+
+type BaseInputWrapperProps = Pick<BaseInputProps, 'isDisabled' | 'validationState'> & {
+  isFocused?: boolean;
+};
+
+const StyledBaseInputWrapper = styled(Box)<BaseInputWrapperProps>((props) => ({
+  ...getInputBackgroundAndBorderStyles({
+    theme: props.theme,
+    isFocused: props.isFocused,
+    isDisabled: props.isDisabled,
+    validationState: props.validationState,
+  }),
+  ':focus-within':
```

Comment:
Tho focus within's browser compatibility is good now. Maybe still will be nice to have a [useFocusWithin](
https://github.com/razorpay/blade/pull/544/commits/7a537169174fb57246900d6ddeaa9cfe79995640#diff-96c5d05c18b47c71d74c2ac0a77194eeda3ac2c505c58bd0d684754e062e34fc) hook which will work seamlessly on native too. 

I had an usecase in Checkbox too:  https://github.com/razorpay/blade/pull/544#discussion_r928864831

Reply:
That's exactly why I'm talking about useFocusWithin. It also does it by checking the focus state. 
But lets assess this in another PR.